### PR TITLE
CLN: index related attributes on Series/DataFrame

### PR DESCRIPTION
**Pull request by topper-123**
Followup to #31126. IMO the current approach to adding the index related class attributes is too indirect and therefore unnecessary difficult to follow. Just adding the class attributes directly on the relevant class makes reading the code easier, IMO.

Notice that the types are already defined in pandas/core/generic.py:304-316.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -4565,6 +4565,13 @@ def to_period(self, freq=None, copy=True) -> "Series":
     index: "Index" = properties.AxisProperty(
         axis=0, doc="The index (axis labels) of the Series."
     )
+    _AXIS_ORDERS = ["index"]
+    _AXIS_NUMBERS = {"index": 0}
+    _AXIS_NAMES = {0: "index"}
+    _AXIS_REVERSED = False
+    _info_axis_number = 0
+    _info_axis_name = _AXIS_ORDERS[_info_axis_number]
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: i _think_ should either be a property in NDFrame or just be explicit, i.e `_info_axis_name = "index"`

File: pandas/core/series.py [hunk 2]
```diff
@@ -4565,6 +4565,13 @@ def to_period(self, freq=None, copy=True) -> "Series":
...
+    _AXIS_LEN = len(_AXIS_ORDERS)
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: same

**Review by simonjayhawkins — APPROVED** [review]
Thanks @topper-123 generally lgtm. couple of comments. not blockers.

**Comment by topper-123** [comment]
Ok, I've made them strings/ints.

**Comment by WillAyd** [comment]
Yea definitely find this easier to follow as well

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @topper-123

**Pull request merged**